src/views/common/Animation/AnimationControls/PlaybackCropWindowButton.tsx

```diff
@@ -1,23 +1,20 @@
 import { faArrowsLeftRightToLine } from '@fortawesome/free-solid-svg-icons'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import React, { useCallback, useMemo } from 'react'
-import { AnimationStateDispatcher } from 'views/common/Animation/AnimationStateReducer'
+import { useMemo } from 'react'
+import { PlaybackButtonLogicCallback } from './PlaybackControlButtonLogic'
 
 
 export const CROP_BUTTON = "cropButton"
 
 type PlaybackCropWindowButtonProps = {
-    dispatch: AnimationStateDispatcher<any>
+    cropWindowHandler: PlaybackButtonLogicCallback
```

Comment:
Individual buttons can now depend on the specific logic they require, rather than taking the entire state dispatcher and coding the logic in the button UI code.